addons/hr/models/hr_employee.py

```diff
@@ -360,10 +360,32 @@ def _onchange_timezone(self):
             self.tz = self.resource_calendar_id.tz
 
     def _sync_user(self, user, employee_has_image=False):
-        vals = dict(
-            work_contact_id=user.partner_id.id,
-            user_id=user.id,
-        )
+        if not user:
+            existing_partner = self.env['res.partner'].search([
+                ('user_id', '=', False),
+                ('email', '=', self.work_email),
+                ('name', '=', self.name),
+                ], order='id asc', limit=1)
```

Comment:
why not simply get `employee.work_contact_id`?

Reply:
Hi @Bertrand2, thank you for reviewing. 
Previously when we empty the `user_id` on the employee record, it would also empty the `work_contact_id`. However, the `work_email` and `work_phone` would remain. 
In the `test_employee_update_work_contact_id` test, another employee is assigned an existing user which will then take on that `work_contact_id`. I wanted to try to separate employee from the old user and old res.partner but keeping a way to have the employee tied to a partner record.